### PR TITLE
Added Biparate  Matching Algorithms

### PR DESCRIPTION
**Pull request by SarojArchana**
Fixes ##11140

### CODE REVIEW

File: graphs/maximum_bipartite_matching.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    This class implements finding maximal Bipartite matching using DFS.
+
+    Usage:
+    >>> bipartite_graph = [
+    ...     [0, 1, 1, 0, 0, 0],
+    ...     [1, 0, 0, 1, 0, 0],
+    ...     [0, 0, 1, 0, 0, 0],
+    ...     [0, 0, 1, 1, 0, 0],
+    ...     [0, 0, 0, 0, 0, 0],
+    ...     [0, 0, 0, 0, 0, 1]
```
Comment (imSanko) [inline graphs/maximum_bipartite_matching.py:12]: We should need to add a couple more doctests ?

**Review by imSanko — COMMENTED** [review]
There ain't much more doctests to make it

File: graphs/maximum_bipartite_matching.py (hunk 1 above)
Comment (SarojArchana) [reply graphs/maximum_bipartite_matching.py]: Hi @imSanko I have added more test case please review now.

Comment (imSanko) [reply graphs/maximum_bipartite_matching.py]: Nice work ✅

**Review by imSanko — APPROVED** [review]

File: graphs/maximum_bipartite_matching.py (hunk 1 above)
Comment (SarojArchana) [reply graphs/maximum_bipartite_matching.py]: @imSanko Thank you for approving, 
I am also interested in solving other open issues and working on it.

Comment (SarojArchana) [reply graphs/maximum_bipartite_matching.py]: @tianyizheng02  can you review the PR? I have made the changes suggested by @imSanko

Comment (SarojArchana) [reply graphs/maximum_bipartite_matching.py]: @imSanko it is still showing "imSanko was requested for review" Can you please help to Review and merge it.

**Comment by cclauss** [comment]
How is this different from/better than https://github.com/search?q=repo%3ATheAlgorithms%2FPython%20Bipartite ?

**Review by cclauss — CHANGES_REQUESTED** [review]
How is this different from/better than https://github.com/search?q=repo%3ATheAlgorithms%2FPython%20Bipartite ?

**Comment by SarojArchana** [comment]
@cclauss 
> How is this different from/better than https://github.com/search?q=repo%3ATheAlgorithms%2FPython%20Bipartite ?

The Three algorithm you mentioned are for
1.graphs/check_bipatrite.py : To check if the graph is bipartite or not
2. graphs/gale_shapley_bigraph.py: An algorithm for finding a solution to the stable matching problem
3.  graphs/dinic.py :is a code for strongly polynomial algorithm for computing the maximum flow in a flow network.

The  implementation provided by me is for  very different task. It is implementation of finding **Maximum Bipartite Matching**.This involves finding the largest possible matching in a bipartite graph. A bipartite graph is one whose vertices can be divided into two disjoint sets such that every edge connects a vertex in one set to a vertex in the other set.

It has multiple application to list few 

1. Online Dating Platforms: Algorithms use bipartite graphs to match users based on shared interests, aiming to pair individuals who have expressed mutual interest in each other.
2. Music and Movie Recommendations: Entertainment platforms utilize bipartite matching to recommend content to users by pairing their preferences with available music or movies, maximizing user satisfaction.
3. Resource Allocation in Volunteer Organizations: Bipartite matching helps efficiently assign volunteers to tasks based on their skills and interests, ensuring effective utilization of skills while meeting project needs.

**Review by SarojArchana — COMMENTED** [review]
> How is this different from/better than https://github.com/search?q=repo%3ATheAlgorithms%2FPython%20Bipartite ?

The Three algorithm you mentioned are for
1.graphs/check_bipatrite.py : To check if the graph is bipartite or not
2. graphs/gale_shapley_bigraph.py: An algorithm for finding a solution to the stable matching problem
3.  graphs/dinic.py :is a code for strongly polynomial algorithm for computing the maximum flow in a flow network.

The  implementation provided by me is for  very different task. It is implementation of finding **Maximum Bipartite Matching**.This involves finding the largest possible matching in a bipartite graph. A bipartite graph is one whose vertices can be divided into two disjoint sets such that every edge connects a vertex in one set to a vertex in the other set.

It has multiple application to list few 

1. Online Dating Platforms: Algorithms use bipartite graphs to match users based on shared interests, aiming to pair individuals who have expressed mutual interest in each other.
2. 
3. Music and Movie Recommendations: Entertainment platforms utilize bipartite matching to recommend content to users by pairing their preferences with available music or movies, maximizing user satisfaction.
4. 
5. Resource Allocation in Volunteer Organizations: Bipartite matching helps efficiently assign volunteers to tasks based on their skills and interests, ensuring effective utilization of skills while meeting project needs.

**Comment by SarojArchana** [comment]
@cclauss could you please help to review my pull request.

**Comment by SarojArchana** [comment]
@cclauss could you please help to review my pull request.

**Comment by cclauss** [comment]
Closing awaiting_changes PRs to prepare for Hacktoberfest 2024

**Pull request closed without merging**